digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/process/SevenRestrictionsLimitationValidator.scala

```diff
@@ -0,0 +1,103 @@
+      } else {
+        val nearestAsset = assets.maxBy(_.endMeasure)
+        (nearestAsset, GeometryUtils.geometryLength(roadLink.geometry) - nearestAsset.endMeasure)
+      }
+    }
+
+    val assetOnLink = assets.filter(_.linkId == roadLink.linkId)
+    if (assetOnLink.nonEmpty && assetDistance(assetOnLink)._2 + distance <= radiusDistance) {
+      Seq(assetDistance(assets)._1)
+    } else
+      Seq()
+  }
+
+  def getAssetValue(asset: PersistedLinearAsset): String = {
+    asset.value match {
+      case Some(NumericValue(intValue)) => intValue.toString
+      case _ => ""
+    }
+  }
+
+  override def getAsset(roadLink: Seq[RoadLink]): Seq[AssetType] = {
+    dao.fetchLinearAssetsByLinkIds(assetTypeInfo.typeId, roadLink.map(_.linkId), LinearAssetTypes.numericValuePropertyId, false).filterNot(_.expired)
+  }
+
+  override def verifyAsset(assets: Seq[PersistedLinearAsset], roadLink: RoadLink, trafficSign: PersistedTrafficSign): Set[Inaccurate] = {
+    assets.flatMap { asset =>
+
+      if (!comparingAssetAndTrafficValue(asset, trafficSign))
+        Seq(Inaccurate(Some(asset.id), None, roadLink.municipalityCode, roadLink.administrativeClass))
+      else
+        Seq()
+    }.toSet
+  }
+
+  override def reprocessRelevantTrafficSigns(assetInfo: AssetValidatorInfo): Unit = {
+    if (assetInfo.ids.toSeq.nonEmpty) {
+      withDynTransaction {
+
+        val assets = dao.fetchAssetsWithTextualValuesByIds(assetInfo.ids, LinearAssetTypes.getValuePropertyId(assetTypeInfo.typeId)).filterNot(_.expired)
+        val roadLinks = roadLinkService.getRoadLinksAndComplementariesFromVVH(assets.map(_.linkId).toSet, newTransaction = false).filterNot(_.administrativeClass == Private)
+
+        assets.foreach { asset =>
+          roadLinks.find(_.linkId == asset.linkId) match {
+            case Some(roadLink) =>
+              val assetGeometry = GeometryUtils.truncateGeometry2D(roadLink.geometry, asset.startMeasure, asset.endMeasure)
+              val (first, last) = GeometryUtils.geometryEndpoints(assetGeometry)
+
+              val trafficSingsByRadius: Set[PersistedTrafficSign] = getPointOfInterest(first, last, SideCode.apply(asset.sideCode)).flatMap { position =>
+                splitBothDirectionTrafficSignInTwo(trafficSignService.getTrafficSignByRadius(position, radiusDistance) ++ trafficSignService.getTrafficSign(Seq(asset.linkId)))
+                  .filter(sign => allowedTrafficSign.contains(TrafficSignType.apply(getTrafficSignsProperties(sign, "trafficSigns_type").get.propertyValue.toInt)))
+                  .filterNot(_.floating)
+              }.toSet
+              val allLinkIds = assetInfo.newLinkIds ++ trafficSingsByRadius.map(_.linkId)
+
+              inaccurateAssetDAO.deleteInaccurateAssetByIds(assetInfo.ids)
+              if(allLinkIds.nonEmpty)
+                inaccurateAssetDAO.deleteInaccurateAssetByLinkIds(allLinkIds, assetTypeInfo.typeId)
+
+              trafficSingsByRadius.foreach { trafficSign =>
```

Comment:
This is being done in every validator, maybe could be created a method on the parent and be called